### PR TITLE
Proposal: Add Attributes to H5Easy

### PR DESCRIPTION
**Pull request by tdegeus**
This is a proposal for the extension of the H5Easy API. It would allow:

```cpp
#include <Eigen/Eigen>
#include <highfive/H5Easy.hpp>

int main()
{
    Eigen::Vector3d a;
    a(0) = 1.0;
    a(1) = 2.0;
    a(2) = 3.0;

    H5Easy::File data("test.h5", H5Easy::File::Overwrite);

    H5Easy::dump(data, "/a", a);
    H5Easy::dump_attribute(data, "/a", "desc", std::string("Foobar"));
    std::cout << H5Easy::load_attribute<std::string>(data, "/a", "desc") << std::endl;
}
```

Comments would be great!

### CODE REVIEW

**Comment by alkino** [comment]
Could you not used the same function with an overload if `data` is of type `H5::Attribute`?

File: include/highfive/H5Easy.hpp [hunk 1]
```diff
@@ -67,21 +68,21 @@ inline size_t getSize(const File& file, const std::string& path);
 ///
 /// \brief Get the shape of an existing DataSet in an readable file.
 ///
-/// \param file A readable opened file
+/// \param file Readable opened HDF5 file
```
Comment (matz-e) [inline include/highfive/H5Easy.hpp]:
```suggestion
/// \param file Open HDF5 file, readable
```

File: include/highfive/H5Easy.hpp [hunk 2]
```diff
@@ -67,21 +68,21 @@ inline size_t getSize(const File& file, const std::string& path);
...
 ///
 /// \return the shape of the DataSet
 inline std::vector<size_t> getShape(const File& file, const std::string& path);
 
 ///
-/// \brief Write an Eigen matrix to a new DataSet in an open HDF5 file.
+/// \brief Write an Eigen matrix to a DataSet in an open HDF5 file.
 ///
-/// \param file Writeable opened file
+/// \param file Writeable opened HDF5 file
```
Comment (matz-e) [inline include/highfive/H5Easy.hpp]:
```suggestion
/// \param file Open HDF5 file, writable
```

File: include/highfive/H5Easy.hpp [hunk 3]
```diff
@@ -58,7 +59,7 @@ enum class DumpMode
 ///
 /// \brief Get the size of an existing DataSet in an open HDF5 file.
 ///
-/// \param file A readable opened file
+/// \param file Readable opened HDF5 file
```
Comment (matz-e) [inline include/highfive/H5Easy.hpp]:
```suggestion
/// \param file Open HDF5 file, readable
```

File: include/highfive/H5Easy.hpp [hunk 4]
```diff
@@ -92,14 +93,14 @@ inline DataSet dump(File& file,
 #endif
 
 ///
-/// \brief Write "xt::xarray<T>" to a new DataSet in an open HDF5 file.
+/// \brief Write "xt::xarray<T>" to a DataSet in an open HDF5 file.
 ///
-/// \param file A writeable opened HDF5 file
+/// \param file Writeable opened HDF5 file
```
Comment (matz-e) [inline include/highfive/H5Easy.hpp]:
```suggestion
/// \param file Open HDF5 file, writable
```

File: include/highfive/H5Easy.hpp [hunk 5]
```diff
@@ -110,14 +111,14 @@ inline DataSet dump(File& file,
...
 
 ///
-/// \brief Write "xt::xtensor<T,rank>" to a new DataSet in an open HDF5 file.
+/// \brief Write "xt::xtensor<T,rank>" to a DataSet in an open HDF5 file.
 ///
-/// \param file opened File (has to be writeable)
-/// \param path path of the DataSet
-/// \param data the data to write
-/// \param mode write mode
+/// \param file Writeable opened HDF5 file
```
Comment (matz-e) [inline include/highfive/H5Easy.hpp]:
```suggestion
/// \param file Open HDF5 file, writable
```

**Comment by tdegeus** [comment]
> Could you not used the same function with an overload if `data` is of type `H5::Attribute`?

I'm not entirely sure what you mean. Are you talking about the API or the internals?

**Comment by ferdonline** [comment]
Hi Tom. Thanks for the contribution.
To my mind the API looks fine, mimics that of `H5Easy::dump/load`.
The thing that is missing are some tests 😄

**Comment by tdegeus** [comment]
Now that we agree on the API I will finish this PR, incl. tests/examples

**Comment by tdegeus** [comment]
Awaiting #314 for a clean implementation

**Comment by tdegeus** [comment]
API furthermore depends on #330

**Comment by tdegeus** [comment]
Superseded by #337

**Pull request closed without merging**